File: test/ch21-benchmarks/bench-007-fibonacci.py
#!/usr/bin/env python3
# BENCH-007: Fibonacci recursive (n=20)
# Python baseline implementation
# Note: Reduced from n=30 to n=20 for practical timing in interpreted modes

def fibonacci(n):
    """Calculate nth Fibonacci number recursively"""
    if n <= 1:
        return n
    return fibonacci(n - 1) + fibonacci(n - 2)

def main():
    result = fibonacci(20)
    print(f"fib(20) = {result}")
    assert result == 6765, f"Expected 6765, got {result}"

if __name__ == "__main__":
    main()

File: test/ch21-benchmarks/bench-007-fibonacci.c
// BENCH-007: Fibonacci recursive (n=20) - C
// Tests: function call overhead, stack management

#include <stdio.h>

int fibonacci(int n) {
    if (n <= 1) {
        return n;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

int main() {
    int result = fibonacci(20);
    // Silent for benchmarking
    return 0;
}
